### PR TITLE
Added E2E tests for k8s components and mtls 

### LINKED ISSUE #671: MTLS command on error exits with success
## Expected Behavior

MTLS command on error exits with success

## Actual Behavior

MTLS command on error exits with an error

## Steps to Reproduce the Problem

on a non-dapr installed cluster ... run `dapr mtls -k`

## Release Note

RELEASE NOTE: **RESOLVED** MTLS command on error exits with success

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Added components check and mtls check to e2e tests.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #662 
closes #671 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: cmd/mtls.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ dapr mtls -k
 		enabled, err := kubernetes.IsMTLSEnabled()
 		if err != nil {
 			print.FailureStatusEvent(os.Stdout, fmt.Sprintf("error checking mTLS: %s", err))
-			return
+			os.Exit(1)
```
Comment (mukundansundar) [inline cmd/mtls.go:32]: On error exit with error also

File: tests/e2e/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -203,6 +214,98 @@ func testInstall(haMode bool) func(t *testing.T) {
...
+		_, err = os.Stat("./certs/issuer.key")
+		require.NoError(t, err, "expected file to exist")
+		err = os.RemoveAll("./certs")
+		if err != nil {
+			t.Logf("error on removing local certs directory %s", err.Error())
+		}
+	}
+}
+
+func testComponents(isDaprInstalled bool) func(t *testing.T) {
```
Comment (wcs1only) [inline tests/e2e/kubernetes/kubernetes_test.go:266]: nit: This construction is a little weird to me. The isDaprInstalled case and the !isDaprInstalled case don't actually share any code? Doesn't it make things a little clearer if they were just separate functions?

**Review by wcs1only — APPROVED** [review]
Minor nitpick that you can incorporate or ignore at your discretion.

File: tests/e2e/kubernetes/kubernetes_test.go (hunk 2 above)
Comment (mukundansundar) [reply tests/e2e/kubernetes/kubernetes_test.go:266]: Will be incorporating those changes in #665 . Already fixed it as part of that code.....

**Pull request merged**